lms/djangoapps/program_enrollments/rest_api/v1/utils.py

```diff
@@ -111,10 +168,143 @@ def wrapped_function(self, request, **kwargs):
+        return view_func(self, *args, **kwargs)
     return wrapped_function
 
 
+def verify_user_enrolled_in_program(view_func):
+    """
+    Raised PermissionDenied if the `target_user` is not enrolled in the program.
+
+    Expects to be used within a UserProgramViewMixin subclass.
+    """
+    @wraps(view_func)
+    def wrapped_function(self, *args, **kwargs):
+        """
+        Wraps the given view_function.
+        """
+        user_enrollment_qs = fetch_program_enrollments(
+            program_uuid=self.program_uuid,
+            users={self.target_user},
+            program_enrollment_statuses={ProgramEnrollmentStatuses.ENROLLED},
+        )
+        if not user_enrollment_qs.exists():
+            raise PermissionDenied
+        return view_func(self, *args, **kwargs)
+    return wrapped_function
+
+
+def get_enrollments_for_courses_in_program(user, program):
+    """
+    Get a user's active enrollments for course runs with the given program.
+
+    Note that this is distinct from the user's *program course enrollments*,
+    which refers to courses that were enrollmed in *through* a program.
+
+    In the case of this function, the course runs themselves must be part of the
+    program, but the enrollments do not need to be associated with a program enrollment.
+
+    Arguments:
+        user (User)
+        program (dict)
+
+    Returns QuerySet[CourseEnrollment]
+    """
+    course_keys = [
+        CourseKey.from_string(key)
+        for key in course_run_keys_for_program(program)
+    ]
+    return CourseEnrollment.objects.filter(
+        user=user,
+        course_id__in=course_keys,
+        mode__in=[CourseMode.VERIFIED, CourseMode.MASTERS],
+        is_active=True,
+    )
+
+
+def get_enrollment_overviews(user, program, enrollments, request):
+    """
+    Get a list of overviews for a user's enrollments in a course runs within a program.
+
+    Arguments:
+        user (User)
+        program (dict)
+        enrollments (iterable[CourseEnrollment])
+        request (HttpRequest): Source HTTP request. Needed for URL generation.
+
+    Returns list[dict]
+    """
+    overviews_by_course_key = {
+        enrollment.course.id: enrollment.course for enrollment in enrollments
+    }
+    course_keys = list(overviews_by_course_key.keys())
+    certficates_by_course_key = get_certificates_for_user_by_course_keys(user, course_keys)
+    resume_urls_by_course_key = get_resume_urls_for_enrollments(user, enrollments)
+    return [
+        get_enrollment_overview(
+            user=user,
+            program=program,
+            course_overview=overviews_by_course_key[enrollment.course_id],
+            certificate_info=certficates_by_course_key.get(enrollment.course_id, {}),
+            relative_resume_url=resume_urls_by_course_key.get(enrollment.course_id),
+            request=request,
+        )
+        for enrollment in enrollments
+    ]
+
+
+def get_enrollment_overview(
```

Comment:
The name of this function is too similar to the name of `get_enrollment_overviews`. Can we be a bit verbose so we know the distinction?

Reply:
Good point. Agreed. I'll change it to `get_single_enrollment_overview`.